### ISSUE TITLE
Unable to reproduce results

### ISSUE THREAD

**Issue by pclucas14** [LiqunChen0606/FM-GAN#5]
Hi, 

I am trying to reproduce your results so that I can cite your paper as a baseline. However, after running `python autoencoder.py` followed by `python textGAN_news.py` the model collapses after 1 adversarial epoch and always outputs the same word. Would it be possible for you to provide a script and hyperparameters to clearly reproduce your results ?

Thanks, 
Lucas

**Comment by LiqunChen0606** [comment]
Hi Lucas,

I am very sorry for the confusing here. Please test the code textGAN_news2.py, not the textGAN_news.py.
Because textGAN_news.py is the code I want to compare my model with MMD textGAN, and sinkhorn textGAN and Gumbel-softmax textGAN.

If you search for G_loss andD_loss, you can find the major difference between these two codes.

Best,
Liqun

**Comment by pclucas14** [comment]
Hi Liqun, 

thanks for the fast reply. Things are running now. How did you cross-validate your GAN model ? Do I just take the last GAN saved after running textGAN_news2 ? 

Thanks, 
Lucas

**Comment by LiqunChen0606** [comment]
Hi Lucas,

In my case, I created a development dataset from the training data. I stored the model with the best BLEU score for the dev.

Best,
Liqun

**Comment by pclucas14** [comment]
Hi again, 

I am still unable to reproduce the results. Throughout the whole GAN training, `Val BLEU (2,3,4): 0.0 0.0 0.0`. I thought this was a mistake, but no, the values are not 0 but super super tiny. 
What do you suggest I do ? Is this normal ?

**Comment by LiqunChen0606** [comment]
Hi Lucas,

I find my saved model this afternoon. You can download all of them from the dropbox link here:
https://www.dropbox.com/s/frestvnzpl4csqq/news_128_dim_cnn_rnn_s0.data-00000-of-00001?dl=0
https://www.dropbox.com/s/za3my3w2rg8jjec/news_128_dim_cnn_rnn_s0.index?dl=0
https://www.dropbox.com/s/0d8712aq1eanxzv/news_128_dim_cnn_rnn_s0.meta?dl=0

Do you load the model from the pre-trained MLE model for training?
I never encounter this problem before. Hopefully the uploaded model can solve the problem for you.

Best

**Comment by pclucas14** [comment]
Hi Liqun,

Thank you very much for uploading your models! I was indeed loading the model obtained 99 epochs of running autoencoder.py (I was to play with the restore_from_save method to get things to work)

I'll update you if there is anything else.
Cheers, 
Lucas

**Comment by pclucas14** [comment]
Everything worked out! I was able to get BLEU-5 and SBLEU-5 scores for news : 
I got 0.3719 and 0.996. The BLEU-5 score is pretty close to the one in the paper (0.399). 

Just to double check with you, here is what I did : 
1) I passed the `is_sampling=True` when running `lstm_decoder_embedding`
2) I sampled and saved 10 * 1000 = 10k sentences
3) I converted them using `convert_news.py`
4) I ran `selfbleu_arxiv.py` and   `testBleu_arxiv.py`, which I modified to get BLEU-5 and SBLEU-5 scores. 

I will be reporting these results. Please let me know if you have anything to add. 

Best, 
Lucas

**Comment by pclucas14** [comment]
I'm having trouble reproducing the S-BLEU results of this model on BLUE. Is it possible you were sampling at different temperatures for both models ?

**Comment by LiqunChen0606** [comment]
Hi Lucas,

I think maybe because 

(i)  I only use 2000 or 5000 (I can't remember clearly) generated samples for testing BLEU and self-BLEU score. (possibly not the main reason)

(ii) the temperatures are different. Because when testing the code, the temperature does not need to be large. Specifically,  line 154 in textGAN_news2.py:       prob = [tf.nn.softmax(l*opt.L) for l in logits], you don't need the opt.L for testing, otherwise it will definitely overfitting the data 
(or you can have a small opt.L for better results, I cannot remember what L I use, I think you can test opt.L=1, 10 and 50. ). 

Best,
Liqun

**Comment by pclucas14** [comment]
Hi Liqun, 

(i) I sampled 10k tokens for both BLEU and self-BLEU to be consistent with the litterature. 

(ii) Just to confirm, you used different temperatures for testing BLEU vs testing SBLEU ? Or did you use different temperatures from training vs testing ?

**Comment by LiqunChen0606** [comment]
Hi Lucas,

Different temp for training and testing.

Best,

**Comment by pclucas14** [comment]
Okay! I  will play with the opt.L parameter and get back to you.

**Comment by pclucas14** [comment]
So I am still unable to reproduce your results. While the BLEU scores are close to what you reported in your paper, the self-BLEU scores are far off. I played with multiple temperatures, and still I cannot match the results. 
I think it would be best if you could double check this; It is quite hard for me to trust the results in the paper if I cannot even reproduce them with a trained model.

Nevertheless, I would like to thank you for your prompt replies to my questions and for sharing your final model. 

Best, 
Lucas

**Comment by LiqunChen0606** [comment]
Hi Lucas,

Thanks for you update. I will double check the trianed model and code.

And I want to clarify that I did not report Self BLEU-5 for all dataset in my paper, besides, I use 2k sentences for BLEU and self BLEU calculation instead of 10k. It is very possible to have different results here.

Best,
Liqun

**Comment by pclucas14** [comment]
Hi Liqun, 

I was referring to self-BLEU2-4. I was under the impression that, for other models, you reported the results from their paper. It seems that it is not the case. If you reran other models with self-Bleu using 2k sentences, then that is fine.

**Comment by LiqunChen0606** [comment]
Hi Lucas,

If you check this Texygen paper: https://arxiv.org/pdf/1802.01886.pdf, you can find my reported results are very different from theirs, because I re-ran all the model under the same setup for fair comparison.

Best,
Liqun

**Comment by pclucas14** [comment]
Great! I think everything works under this new evaluation framework. 

Thanks again, 
Lucas

**Comment by z289260161** [comment]
Hi Lucas,

@pclucas14 
I encountered the same "mode collapse" problem after I ran "autoencoder.py" and "textGAN_news2.py".  It seems that you overcame the problem and reproduced similar results from scratch.

I'm wondering whether the author's pre-trained model is necessary. If not,  how do you modify the code?

Best,
Wenqing

**Comment by pclucas14** [comment]
Hi, 

I was not able to train the model from scratch, I used the pretrained one. 

Best, 
Lucas

**Comment by AliceZhang2016** [comment]
Hi, 

I am trying to reproduce results with WMT dataset and I failed with running textGAN_news2.py with the model obtained 99 epochs after running autoencoder.py, Val BLEU(2,3,4) are always 0.0 0.0 0.0 as Lucas said in this issue. 

However, when I run textGAN_news2.py with your upload pretrained model, it still not works. All the loss except GAN loss turns to nan after epoch 0 and epoch1.

Have you ever encountered such problem?
Do I need to modify some parts in the original code of textGAN_news2.py? 

Thank you very much!

Here is part of my log data:
===========================
Starting epoch 0
Validation d_loss -0.139757, g_loss 0.140931  mean_dist 0.000528
Sent:To be honest , we need to understand that the public , which is a problem with the people who are doing , and it ' s not just a small number .
MMD loss 0.003010, GAN loss 0.691689
Val BLEU (2,3,4): 0.0 0.0 0.0
Validation d_loss -0.128163, g_loss 0.139691  mean_dist 0.000534
Sent:The fourth year in the United States , Mr Trump ' s transition team will begin investigating .
MMD loss 0.003030, GAN loss 0.691665
Val BLEU (2,3,4): 0.0 0.0 0.0
Starting epoch 1
Validation d_loss -0.139057, g_loss 0.125936  mean_dist 0.000576
Sent:They are not paying the money , but they are still getting a lot of money from them ," he said .
MMD loss 0.003257, GAN loss 0.691662
Val BLEU (2,3,4): 0.0 0.0 0.0
Validation d_loss -0.226418, g_loss 0.131956  mean_dist 0.001258
Sent:The government will be able to meet the final case , but also confirmed that the government had been able to return to .
MMD loss 0.007109, GAN loss 0.692523
Val BLEU (2,3,4): 0.0 0.0 0.0
Validation d_loss nan, g_loss nan  mean_dist nan
Sent:
MMD loss nan, GAN loss 0.691565
Val BLEU (2,3,4): 0.0 0.0 0.0
Starting epoch 2
Validation d_loss nan, g_loss nan  mean_dist nan
Sent:
MMD loss nan, GAN loss 0.691565
===========================

**Comment by gr8joo** [comment]
I have exactly same symptoms as @AliceZhang2016 

@pclucas14 Did you resolved this issue?

or anyone else did?
